docs/git-guide.md

```diff
@@ -0,0 +1,1052 @@
+
+```
+$ git add test3.txt
+On branch issue-1234
+Changes to be committed:
+  (use "git reset HEAD <file>..." to unstage)
+
+        new file:   test3.txt
+
+$ git reset HEAD test3.txt
+$ git status
+On branch issue-1234
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+        test3.txt
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+#### Stage deletions with git rm
+
+To remove existing files from your repo, use `git rm` ([documentation][gitbook-rm]). This command can either stage the file for removal from your repository AND delete it from your working directory or just stage the file for deletion and leave it in your working directory.
+
+To stage a file for deletion and **remove** it from your working directory, use `git rm <filename>`:
+
+```
+$ git rm test.txt
+rm 'test.txt'
+
+$ git status
+On branch issue-1234
+Changes to be committed:
+  (use "git reset HEAD <file>..." to unstage)
+
+        deleted:    test.txt
+
+$ ls test.txt
+ls: No such file or directory
+```
+
+To stage a file for deletion and **keep** it in your working directory, use `git rm --cached <filename>`:
+
+```
+$ git rm --cached test2.txt
+rm 'test2.txt'
+
+$ git status
+On branch issue-1234
+Changes to be committed:
+  (use "git reset HEAD <file>..." to unstage)
+
+        deleted:    test2.txt
+
+$ ls test2.txt
+test2.txt
+```
+
+If you stage a file for deletion with the `--cached` option, and haven't yet run `git commit`, you can undo it with `git reset HEAD <filename>`:
+
+```
+$ git reset HEAD test2.txt
+```
+
+Unfortunately, you can't restore a file deleted with `git rm` if you didn't use the `--cache` option. However, `git rm` only deletes files it knows about. Files you have never added to git won't be deleted.
+
+### Commit changes
+
+When you've staged all your changes, you're ready to commit. You can do this with `git commit -m "My commit message."` to include a commit message.
+
+Here's an example of commiting with the `-m` for a one-line commit message:
+
+```
+$ git commit -m "Add a test commit for docs."
+[issue-123 173e17a] Add a test commit for docs.
+ 1 file changed, 1 insertion(+)
+ create mode 100644 newfile.py
+```
+
+You can also use `git commit` without the `-m` option and your editor to open, allowing you to easily draft a multi-line commit message.
+
+How long your commit message should be depends on where you are in your work. Use short, one-line messages for commits related to in-progress work makes sense. For a commit that you intend to be final or that encompasses a significant amount or complex work, you should include a longer message.
+
+Keep in mind that your commit should contain a 'minimal coherent idea' and have a quality commit message. See Zulip docs [Commit Discipline][zulip-rtd-commit-discipline] and [Commit messages][zulip-rtd-commit-messages] for details.
```

Comment:
I don't _think_ it needs the "on" to be grammatically correct, but not 100% sure. And if it reads funny as-is, I'm happy to change. (I think of this sentence more like, "See Zulip contributors Tim and Christie for details.")